src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -173,30 +182,100 @@ public DownloadFile exportObservations(
+            // This will hold a list of rows for each study, each list will become a separate file.
+            Map<String, List<Map<String, Object>>> rowsByStudyId = new HashMap<>();
+
+            for (Map<String, Object> row: rowByOUId.values()) {
+                String studyId = studyDbIdByOUId.get((String)row.get(ExperimentObservation.Columns.OBS_UNIT_ID));
+                // Initialize key with empty list if it is not present.
+                if (!rowsByStudyId.containsKey(studyId))
+                {
+                    rowsByStudyId.put(studyId, new ArrayList<Map<String, Object>>());
+                }
+                // Add row to appropriate list in rowsByStudyId.
+                rowsByStudyId.get(studyId).add(row);
+            }
+            List<DownloadFile> files = new ArrayList<>();
+            // Generate a file for each study.
+            for (Map.Entry<String, List<Map<String, Object>>> entry: rowsByStudyId.entrySet()) {
+                StreamedFile streamedFile = writeToStreamedFile(columns, entry.getValue(), fileType, "Experiment Data");
+                String name = makeFileName(experiment, program, studyByDbId.get(entry.getKey()).getStudyName()) + fileType.getExtension();
+                // Add to file list.
+                files.add(new DownloadFile(name, streamedFile));
+            }
+            if (files.size() == 1) {
+                // Don't zip, as there is a single file.
+                downloadFile = files.get(0);
+            }
+            else {
+                // Zip, as there are multiple files.
+                StreamedFile zipFile = zipFiles(files);
+                downloadFile = new DownloadFile(makeZipFileName(experiment, program), zipFile);
+            }
         } else {
-            downloadFile = ExcelWriter.writeToDownload("Experiment Data", columns, exportRows, fileType);
+            List<Map<String, Object>> exportRows = new ArrayList<>(rowByOUId.values());
+            // write export data to requested file format
+            StreamedFile streamedFile = writeToStreamedFile(columns, exportRows, fileType, "Experiment Data");
+            // Set filename.
+            String envFilenameFragment = params.getEnvironments() == null ? "All Environments" : params.getEnvironments();
+            String fileName = makeFileName(experiment, program, envFilenameFragment) + fileType.getExtension();
+            downloadFile = new DownloadFile(fileName, streamedFile);
         }
 
-        String envFilenameFragment = params.getEnvironments() == null ? "All Environments" : params.getEnvironments();
-        String fileName = makeFileName(experiment, program, envFilenameFragment) + fileType.getExtension();
-        return new DownloadFile(fileName, downloadFile);
+        return downloadFile;
+    }
+
+    private StreamedFile writeToStreamedFile(List<Column> columns, List<Map<String, Object>> data, FileType extension, String sheetName) throws IOException {
```

Comment:
The only value passed in for `sheetName` is "Experiment Data'. Can this arg be removed from the method signature?